applications/tube_network/src/statistics.py

```diff
@@ -0,0 +1,234 @@
+# Copyright 2018 Grakn Labs Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import grakn
+import tube_network.settings as settings
+
+def print_to_log(title, content):
+  print(title)
+  print("")
+  print(content)
+  print("\n")
+
+# How many stations do exist?
+def execute_query_1(question, transaction):
+    print_to_log("Question: ", question)
+
+    query = 'compute count in station;'
+
+    print_to_log("Query:", query)
+
+    answer = list(transaction.query(query))[0]
+    number_of_stations = answer.number()
+
+    print("Number of stations: " + str(number_of_stations))
+
+
+# How long is the shortest trip between two stations?
+def execute_query_2(question, transaction):
+    print_to_log("Question: ", question)
+
+    query = 'compute min of duration, in route-section;'
+
+    print_to_log("Query:", query)
+
+    answer = list(transaction.query(query))[0]
+    min_duration = answer.number()
+
+    print("Shortest Trip: " + str(min_duration))
+
+
+# Which is the west most station in London?
+def execute_query_3(question, transaction):
+    print_to_log("Question: ", question)
+
+    query = 'compute min of lat, in station;'
+
+    print_to_log("Query:", query)
+
+    answer = list(transaction.query(query))[0]
+    lat = answer.number()
+
+    query = [
+        'match',
+        '   $sta isa station, has lat $lat, has name $nam;',
+        '   $lat ' + str(lat) + ';',
+        'get $nam;'
+    ]
+
+    print_to_log("Query:", "\n".join(query))
+    query = "".join(query)
+
+    answers = transaction.query(query).collect_concepts()
+    result = [ answer.value() for answer in answers ]
+
+
+    print_to_log("West most stations with " + str(lat) + " are: ", result)
```

Comment:
In fact minimum latitude will determine the northernmost stations! Latitude goes north-south, longitude goes east-west